### ISSUE TITLE
Bug: WS001 false positive - confuses bold syntax with list markers

### ISSUE THREAD

**Issue by raifdmueller** [docToolchain/asciidoc-linter#41]
## Problem

The WS001 rule ("Missing space after the marker") incorrectly flags AsciiDoc bold syntax (`*text*` and `**text**`) as list markers that need a space.

## Example

Given this **valid** AsciiDoc file:

```asciidoc
=== ADR-001: My Decision

*Status:* Accepted (2026-02-11)

*Context:*
The system requires...

*Decision:*
We decided to...

*Consequences:*

**Positive:**
- Benefit 1
- Benefit 2

**Negative:**
- Drawback 1
```

The linter reports:

```
✗ line 3: Missing space after the marker '*'   (*Status:*)
✗ line 5: Missing space after the marker '*'   (*Context:*)
✗ line 8: Missing space after the marker '*'   (*Decision:*)
✗ line 11: Missing space after the marker '*'  (*Consequences:*)
✗ line 13: Missing space after the marker '**' (**Positive:**)
✗ line 17: Missing space after the marker '**' (**Negative:**)
```

## Root Cause

The WS001 rule appears to interpret **any** `*` as a potential list marker, without distinguishing between:

1. **Bold text**: `*word*` or `**word**` (inline formatting)
2. **List items**: `* item` or `** nested item` (block-level, requires space)

## Expected Behavior

WS001 should only flag actual list markers that are:
- At the **start of a line** (after optional whitespace)
- Followed by **non-asterisk characters** without a space

**Valid list marker** (should be flagged):
```asciidoc
*item without space    ← should warn
```

**Valid bold text** (should NOT be flagged):
```asciidoc
*Status:* accepted     ← should NOT warn (inline bold)
**Important:** note    ← should NOT warn (inline bold)
Some *bold* text here  ← should NOT warn (mid-line)
```

## Detection Logic Suggestion

```python
# Pseudo-code for improved WS001
if line.lstrip().startswith('*'):
    # Potential list marker
    marker_match = re.match(r'^(\s*)(\*+)([^\s*])', line)
    if marker_match:
        # This is a list marker without space
        # Flag as WS001
    else:
        # Check if it's bold syntax: *text* or **text**
        if re.match(r'^\s*\*+\w+.*\*+', line):
            # This is bold syntax, not a list
            # Do NOT flag
```

## Impact

**High** - This creates many false positives that:
- Block pre-commit hooks unnecessarily
- Require `--no-verify` workarounds
- Reduce trust in the linter
- Make it hard to distinguish real issues from false positives

## Workaround

Currently, users must:
1. Skip the hook: `git commit --no-verify`
2. Manually review to confirm these are false positives
3. Ignore WS001 warnings

## Test Cases

The linter should pass these valid AsciiDoc files:

**Test 1: Bold field labels (common in ADRs)**
```asciidoc
*Status:* Accepted
*Deciders:* Team
*Decision:* We will...
```

**Test 2: Mixed bold levels**
```asciidoc
**Positive:**
- Item 1

**Negative:**
- Item 2
```

**Test 3: Mid-line bold**
```asciidoc
This is *important* text.
Here is **very important** content.
```

And should fail these:

**Test 4: List without space**
```asciidoc
*Item 1
**Nested item
```

## Related

- We discovered this when integrating asciidoc-linter as a pre-commit hook
- All existing ADR files (ADR-001 through ADR-013) trigger these false positives
- See our repo: docToolchain/dacli (ADR files in `src/docs/arc42/adr/`)
